### PR TITLE
Refactor Brave Ads unit tests/mocks/utils

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/9471

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm ads are delivered as expected
- Confirm ad conversions work as expected

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/client.cc [hunk 1]
```diff
@@ -22,50 +21,60 @@
...
 namespace {
 
-ads::FilteredAdList::iterator FindFilteredAd(
+FilteredAdsList::iterator FindFilteredAd(
     const std::string& creative_instance_id,
-    ads::FilteredAdList* filtered_ad) {
-  return std::find_if(filtered_ad->begin(), filtered_ad->end(),
-      [&creative_instance_id](const ads::FilteredAd& ad) {
+    FilteredAdsList* filtered_ads) {
+  return std::find_if(filtered_ads->begin(), filtered_ads->end(),
```
Comment (emerick) [inline vendor/bat-native-ads/src/bat/ads/internal/client.cc:31]: Maybe `DCHECK(filtered_ads)` here.

File: vendor/bat-native-ads/src/bat/ads/internal/client.cc [hunk 2]
```diff
@@ -22,50 +21,60 @@
...
 }
 
-ads::FilteredCategoryList::iterator FindFilteredCategory(
+FilteredCategoriesList::iterator FindFilteredCategory(
     const std::string& name,
-    ads::FilteredCategoryList* filtered_category) {
-  return std::find_if(filtered_category->begin(), filtered_category->end(),
-      [&name](const ads::FilteredCategory& category) {
+    FilteredCategoriesList* filtered_categories) {
+  return std::find_if(filtered_categories->begin(), filtered_categories->end(),
```
Comment (emerick) [inline vendor/bat-native-ads/src/bat/ads/internal/client.cc:40]: Maybe `DCHECK(filtered_categories)` here.

File: vendor/bat-native-ads/src/bat/ads/internal/client.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/client.cc:31]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/client.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/client.cc:40]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.h [hunk 3]
```diff
@@ -21,13 +21,13 @@
...
 #include "bat/ads/internal/bundle.h"
+#include "bat/ads/internal/classification/page_classifier/page_classifier.h"
 #include "bat/ads/internal/client.h"
 #include "bat/ads/internal/ad_conversions.h"
 #include "bat/ads/internal/ad_notification_result_type.h"
 #include "bat/ads/internal/ad_notifications.h"
 #include "bat/ads/internal/timer.h"
-#include "bat/ads/internal/page_classifier/page_classifier.h"
-#include "bat/ads/internal/purchase_intent/purchase_intent_classifier.h"
+#include "bat/ads/internal/classification/purchase_intent_classifier/purchase_intent_classifier.h"
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_impl.h:30]: nit: include order

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc [hunk 4]
```diff
@@ -88,21 +90,19 @@ AdsImpl::AdsImpl(AdsClient* ads_client)
...
       subdivision_targeting_(std::make_unique<
           SubdivisionTargeting>(ads_client)),
-      frequency_capping_(std::make_unique<FrequencyCapping>(client_.get())),
-      ad_conversions_(std::make_unique<AdConversions>(
-          this, ads_client, client_.get())),
-      page_classifier_(std::make_unique<PageClassifier>(this)),
+      ad_conversions_(std::make_unique<AdConversions>(this)),
+      page_classifier_(std::make_unique<classification::PageClassifier>(this)),
       purchase_intent_classifier_(std::make_unique<PurchaseIntentClassifier>(
           kPurchaseIntentSignalLevel, kPurchaseIntentClassificationThreshold,
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc:101]: should purchase intent sit under the `classification::` namespace as well?

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.h (hunk 3 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.h:30]: Include order will be fixed with https://github.com/brave/brave-browser/issues/9166, thanks

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc (hunk 4 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc:101]: Fixed

**Comment by tmancey** [comment]
CI failed on Android and Windows, restarting

**Pull request merged**